common/src/main/java/io/crate/types/TypeSignature.java

```diff
@@ -143,6 +157,16 @@ public String getBaseTypeName() {
         return DataTypes.ofName(baseTypeName);
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(baseTypeName);
+        out.writeVInt(parameters.size());
+        for (TypeSignature parameter : parameters) {
+            out.writeVInt(parameter.type().ordinal());
```

Comment:
For `Symbol` and `DataType` we've static methods to handle the `ordinal` handling.

```
    public static void toStream(DataType type, StreamOutput out) throws IOException {
        out.writeVInt(type.id());
        type.writeTo(out);
    }
```

Do you think it would make sense to follow that pattern here too?